ingest/src/main/scala/hydra.ingest/http/IngestionEndpoint.scala

```diff
@@ -72,8 +91,45 @@ class IngestionEndpoint[F[_]: Futurable](
           } ~ deleteRequest
         }
       }
+    } ~ pathPrefix("v2" / "topics") {
+      handleExceptions(exceptionHandler) {
+        pathSuffix("records") {
+            put {
+              pubV2
+            }
+        } ~ extractExecutionContext { implicit ec => getTopics }
+      }
     }
 
+  private val showSystemTopics = applicationConfig
+    .getBooleanOpt("transports.kafka.show-system-topics")
+    .getOrElse(false)
+
+  private val filterSystemTopics = (t: String) =>
+    (t.startsWith("_") && showSystemTopics) || !t.startsWith("_")
+
+
+  private def topics(implicit ec: ExecutionContext) : Future[Map[String, Seq[PartitionInfo]]] = {
+    implicit val timeout = Timeout(5 seconds)
+    implicit val cache = GuavaCache[Map[String, Seq[PartitionInfo]]]
```

Comment:
AFAIK this will create a new cache every time this method is called. Probably want that to live somewhere else.

Reply:
This is the way we were originally doing it in TopicMetadataEndpoint.scala. If this is creating a new cache every time, I will replace it in both places.

Reply:
Didn't realize we were doing it that way.. but yeah it is creating a new one with every method call. We could write a test to double check, but that's definitely what it looks like 😬

Reply:
Fair enough. I will work on changing that in both places. I did struggle with the idea of leaving it in the IngestionEndpoint. My opinion is it might be better to change TopicMetadataEndpoint to include both the `GET` for topics and the `GET` for the metadata while including the old code. Thoughts? This would change the BootstrapEndpointV2 as well.

Reply:
Agreed.. I don't love the idea of a `GET` being inside of something called `IngestionEndpoint` or `Bootstrap` or whatever. I think you have the right idea with moving it to `TopicMetadataEndpoint` and maybe create a `TopicMetadataV2Endpoint`?

Reply:
If there is a lot of duplication then just keep them in the same file. We can go in and get rid of the V1 stuff still at a later point.